diff --git a/notifications.manage.inc b/notifications.manage.inc
index 721fbc5..f5d8a23 100644
--- a/notifications.manage.inc
+++ b/notifications.manage.inc
@@ -113,8 +113,8 @@ function notifications_manage_subscriptions_form($account = NULL, $full = TRUE)
     if (!$account) {
       $form['username'][$sub->sid] = array('#value' => theme('username', $sub));
     }
-    $form['send_method'][$sub->sid] = array('#value' => !empty($send_methods[$sub->send_method]) ? $send_methods[$sub->send_method] : $sub->send_method);
-    $form['send_interval'][$sub->sid] = array('#value' => !empty($send_intervals[$sub->send_interval]) ? $send_intervals[$sub->send_interval] : $sub->send_interval);  
+    $form['send_method'][$sub->sid] = array('#value' => !empty($send_methods[$sub->send_method]) ? check_plain($send_methods[$sub->send_method]) : check_plain($sub->send_method));
+    $form['send_interval'][$sub->sid] = array('#value' => !empty($send_intervals[$sub->send_interval]) ? check_plain($send_intervals[$sub->send_interval]) : check_plain($sub->send_interval));  
     $form['status'][$sub->sid] =  array('#value' => $status[$sub->status]);
     // These links will be different depending on whether we are under admin or user account
     if ($account) {
diff --git a/notifications.pages.inc b/notifications.pages.inc
index a2718c7..8a18b62 100644
--- a/notifications.pages.inc
+++ b/notifications.pages.inc
@@ -31,7 +31,7 @@ function notifications_user_overview($form_state, $account) {
     }
   }
   $interval = notifications_user_setting('send_interval', $account);
-  if (isset($send_intervals[$interval])) {      
+  if (isset($send_intervals[$interval])) { 
     $status['interval'] = t('Your default sending interval for new subscriptions is @send_interval', array('@send_interval' => $send_intervals[$interval]));
   }
   
@@ -541,4 +541,4 @@ function theme_notifications_subscription_fields($elements) {
   $output = theme('table', $header, $rows);
   $output .= drupal_render($elements);
   return $output;
-}
\ No newline at end of file
+}